### ISSUE TITLE
Config not updated from 1.4 to 1.6.2

### ISSUE THREAD

**Issue by RealRancor** [znc/znc#1246]
Hi,

recently i have upgraded my ZNC 1.4-2 of Debian Jessie to 1.6.2-2~bpo8+1 from Jessie Backports.

During the upgrade i have killed znc and started it again. After the first start the following is shown which seems to be legit:

```
[ ** ] Found old config from ZNC 1.4. Saving a backup of it.
[ .. ] Creating a config backup...
[ >> ] /path/to/.znc/configs/znc.conf.pre-1.6.2
```

However after the next kill of znc and a restart the following is shown again:

```
[ ** ] Found old config from ZNC 1.4. Saving a backup of it.
[ .. ] Creating a config backup...
[ !! ] File exists
```

When having a look at the .znc/config/znc.conf the version of the file seems still 1.4:

```
Version = 1.4
```

According to http://wiki.znc.in/FAQ#Upgrading_.2F_updating_from_a_previous_release.2C_to_a_new_version the config should be upgraded automatically.

Is there anything else what needs to be done / checked here? I also wasn't able to identify any fixes in the 1.6.3 changelog which might have solved this.

**Comment by Mikaela** [comment]
Did you kill ZNC or something before it had time to save config by itself? What if you `/msg *status saveconfig`

**Comment by RealRancor** [comment]
Hi,

thanks for your reply. znc was running for about 12 hours after the upgrade. Should that be enough for saving the config?

But will also try this saveconfig later this day, thanks for that hint.

**Comment by DarthGandalf** [comment]
ZNC doesn't save config with no reason. It's saved either explicitly by user request, or if you join some channel with chansaver module loaded, or if you change some setting in webadmin.

**Comment by RealRancor** [comment]
@DarthGandalf Ok, thanks. Then it might be useful to add this notice to:

http://wiki.znc.in/FAQ#Upgrading_.2F_updating_from_a_previous_release.2C_to_a_new_version

Currently some one might assume that the config update wen't wrong or similar. From what i have seen there is no notice that ZNC needs to save the config once so it get upgraded.

**Comment by RealRancor** [comment]
Ah, the link is also explicitly saying:

> Your existing configuration will be upgraded to new version when you start it.

which is then a wrong statement according to your posted info.

**Comment by DarthGandalf** [comment]
It meant that ZNC can read old config and upgrades it "in memory". Then when it saves config, such upgraded config will be saved.
But it's confusing, I agree...

**Comment by RealRancor** [comment]
Ok, thats fine. I was just reading that statement and thought something went wrong during the upgrad. :-)

**Comment by DarthGandalf** [comment]
Probably ZNC should be fixed to actually save config after upgrade instead of fixing wiki.
"[ .. ] Creating a config backup... [ !! ] File exists" is confusing regardless of what wiki says.

**Comment by RealRancor** [comment]
Yes, that sounds like something i had expected during an upgrade. Thanks again.

My original issue is solved by saving the config.
